src/node/stage/approved.rs

```diff
@@ -1628,34 +1811,48 @@ impl Approved {
             dkg_key,
             message: dkg_message_bytes.clone(),
         };
-        let message = Message::single_src(&core.full_id, DstLocation::Direct, variant, None, None)?;
+        let message = Message::single_src(
+            &self.node_info.full_id,
+            DstLocation::Direct,
+            variant,
+            None,
+            None,
+        )?;
 
         let recipients: Vec<_> = self
             .dkg_voter
             .participants()
-            .filter(|p2p_node| p2p_node.public_id() != core.id())
+            .filter(|p2p_node| p2p_node.public_id() != self.node_info.full_id.public_id())
             .map(P2pNode::peer_addr)
             .copied()
             .collect();
 
-        core.send_message_to_targets(&recipients, recipients.len(), message.to_bytes());
+        self.comm
+            .send_message_to_targets(&recipients, recipients.len(), message.to_bytes())
+            .await?;
 
-        self.handle_dkg_message(core, dkg_key, dkg_message_bytes, *core.id())
+        // TODO: don't send it over the wire but just internally,
+        // making sure we don't create a recursive call.
+        self.comm
+            .send_message_to_target(&self.comm.our_connection_info()?, message.to_bytes())
+            .await?;
+
+        Ok(())
```

Comment:
Again, just call `handle_dkg_message` and don't worry about infinite recursion. It won't happen.

Reply:
I put the `#[async_recursion]`, and kept a TODO to remove the recursion eventually as discussed before.